kong/db/dao/targets.lua

```diff
@@ -46,6 +47,15 @@ function _TARGETS:insert(entity, options)
     entity.target = formatted_target
   end
 
+  local workspace = workspaces.get_workspace_id()
+  local opts = { nulls = true, workspace = workspace }
+  for existent in self:each_for_upstream(entity.upstream, nil, opts) do
+    if existent.target == entity.target then
```

Comment:
Might be missing the obvious, but why can't we add the `unique = true` property to the `target` field of the `kong/db/schema/entities/targets.lua` file instead?

Reply:
We can have the same target added to different upstreams, wouldn't the `unique = true` stop us from doing that?

Reply:
whatabout adding cache_key?

Reply:
DIscussed with @bungle and @locao, we decide to postpone this change into a later release because of the complexity of adding `cache_key` with the workspace in consideration. It will not be able to fit in the release schedule.

This PR still made the situation better, so there is no reason to require the `cache_key` change right here right now.